backend/controller/dal/dal.go

```diff
@@ -501,22 +501,51 @@ func (d *DAL) CreateDeployment(ctx context.Context, language string, moduleSchem
 		}
 	}
 
-	// upsert subscriptions
+	// update subscriptions
+	subscriptionNames := []string{}
 	for _, decl := range moduleSchema.Decls {
 		s, ok := decl.(*schema.Subscription)
 		if !ok {
 			continue
 		}
-		err := tx.UpsertSubscription(ctx, sql.UpsertSubscriptionParams{
+		// look for subscribers
+		if _, hasSubscriber := slices.Find(moduleSchema.Decls, func(in schema.Decl) bool {
+			v, ok := in.(*schema.Verb)
+			if !ok {
+				return false
+			}
+			_, subscribes := slices.Find(v.Metadata, func(in schema.Metadata) bool {
+				md, ok := in.(*schema.MetadataSubscriber)
+				if !ok {
+					return false
+				}
+				return md.Name == s.Name
+			})
+			return subscribes
```

Comment:
really nitty nit pick on readability: If you pull this logic into a helper function `hasSubscriber`, you can do:

```
if hasSubscriber(moduleSchema.Decls) {
  // Ignore subscriptions without ...
  continue
}
```

This (1) shortens the for-loop so it's easier to read/comprehend and (2) communicates "look for subscribers" with a func name instead of a comment, which is slightly better for maintainability

Reply:
Yeh agreed, thanks Denise